command/keyring/keyring.go

```diff
@@ -71,7 +75,7 @@ func (c *cmd) Run(args []string) int {
 	}
 
 	// Only accept a single argument
-	found := c.listKeys
+	found := c.listKeys || c.listPrimaryKeys
 	for _, arg := range []string{c.installKey, c.useKey, c.removeKey} {
```

Comment:
List primary keys is a separate action from listKeys , is that right? Should `c.listPrimaryKeys` be part of the slice on 79 instead of OR at the top?

Reply:
This bit of code left me puzzled as well. It determines if the flags are used correctly and only one at a time. `c.listKeys || c.listPrimaryKeys` these are the flags that don't take arguments and `c.installKey, c.useKey, c.removeKey` are the ones that do. This is why `listPrimaryKeys` is in L78. The for loop below checks if the other flags have a key argument.

Reply:
>  It determines if the flags are used correctly and only one at a time.

I think this is right.

> c.listKeys || c.listPrimaryKeys these are the flags that don't take arguments and c.installKey, c.useKey, c.removeKey are the ones that do. This is why listPrimaryKeys is in L78. The for loop below checks if the other flags have a key argument.

Hmm, I'm not sure this is right. Why would it matter if the flags have arguments are not? The arg parsing is handled by the `flags` package well before it gets here.

Looking at how these variables are used below, I think what this is trying to do is ensure that only one of these flags is set. Before this PR that is what was happening. Because there was only one boolean flag I think the author structured it in a bit of a strange way. I think `listPrimaryKeys` needs to be mutually exclusive with the rest of the flags, otherwise the user will not get an error when it is used with `listKeys`, even though they will only get the output for one of those.

I think this should be something like this:

```diff
diff --git a/command/keyring/keyring.go b/command/keyring/keyring.go
index 577c35cce..039b257a3 100644
--- a/command/keyring/keyring.go
+++ b/command/keyring/keyring.go
@@ -74,18 +74,26 @@ func (c *cmd) Run(args []string) int {
 		Ui:           c.UI,
 	}
 
-	// Only accept a single argument
-	found := c.listKeys || c.listPrimaryKeys
-	for _, arg := range []string{c.installKey, c.useKey, c.removeKey} {
-		if found && len(arg) > 0 {
+	// Error if there are flags requesting multiple actions
+	var count int
+	for _, isSet := range []bool{
+		c.listKeys,
+		c.listPrimaryKeys,
+		c.installKey != "",
+		c.useKey != "",
+		c.removeKey != "",
+	} {
+		if isSet {
+			count++
+		}
+		if count > 1 {
 			c.UI.Error("Only a single action is allowed")
 			return 1
 		}
-		found = found || len(arg) > 0
 	}
 
 	// Fail fast if no actionable args were passed
-	if !found {
+	if count == 0 {
 		c.UI.Error(c.Help())
 		return 1
 	}

```